src/server/status/index.js

```diff
@@ -11,20 +11,28 @@ export default function (kbnServer, server, config) {
   }
 
   const wrapAuth = wrapAuthConfig(config.get('status.allowAnonymous'));
-
+  const matchSnapshot = /-SNAPSHOT$/;
   server.route(wrapAuth({
     method: 'GET',
     path: '/api/status',
     handler: function (request, reply) {
-      return reply({
+      const status = {
         name: config.get('server.name'),
-        version: config.get('pkg.version'),
-        buildNum: config.get('pkg.buildNum'),
-        buildSha: config.get('pkg.buildSha'),
         uuid: config.get('server.uuid'),
-        status: kbnServer.status.toJSON(),
-        metrics: kbnServer.metrics
-      });
+        version: {
+          number: config.get('pkg.version').replace(matchSnapshot, ''),
+          build_hash: config.get('pkg.buildSha'),
+          build_number: config.get('pkg.buildNum'),
+          build_snapshot: matchSnapshot.test(config.get('pkg.version'))
+        },
+        status: kbnServer.status.toJSON()
+      };
+
+      if (config.get('status.metrics')) {
```

Comment:
Is there a time which we would not want to return the metrics? This could cause problems with beats consuming it if they can't rely on it being returned.

In 5.x we will need to provide a config to toggle the format introduced in this PR.